src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -198,6 +210,52 @@ private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
                 }
             }
 
+            private void ProcessDocumentOrProjectRemoval(DiagnosticsUpdatedArgs e)
+            {
+                if (e.Solution != null)
```

Comment:
That's a bit subtle: e.Solution == null is the only way to know it's a project removal?